### DISCUSSION TITLE
pytest API for test suites

### DISCUSSION THREAD

**Discussion by AdriMarteau** [Giskard-AI/giskard-oss#1681] [Q&A] [answered]
I am looking at integrating giskard in a CI process and I was wondering if it was compatible with pytest. 
This would help with leveraging CI tools reporting capabilities.

My idea was to use a command like:
```
pytest giskard-tests.py --junitxml=junit/test-results.xml
```

**Comment by luca-martial** [comment, marked as answer]
👋 @AdriMarteau - thanks for sending a question, yes indeed it is possible. We have a section on our docs about this: https://docs.giskard.ai/en/latest/integrations/cicd/pipeline.html

Let us know if you run into any trouble implementing it!

**Reply by AdriMarteau** [reply]
Hi @luca-martial and thanks for your reply! 
Indeed I managed to integrate with my CI but I wanted to push one step further and leverage PyTest API and capacity to create junit reports that are easy to parse.

Thanks again for your help

**Reply by luca-martial** [reply]
@AdriMarteau apologies, I just understood what you meant - actually we'd love to standardize the reporting from the test suite, we could add a `to_junit` method to our `TestSuiteResult`. Would you be interested in contributing to our repo with this addition? It should be pretty straightforward. I've just created a [GH issue](https://github.com/Giskard-AI/giskard/issues/1685) and can assign you!
